### PR TITLE
Newman tobias model

### PR DESCRIPTION
**Pull request by rtimms**
# Description

Adds a new lithium-ion model based on the formulation in _JS Newman and CW Tobias. "Theoretical Analysis of Current Distribution in Porous Electrodes". Journal of The Electrochemical Society,1962_. 

The model assumes uniform (constant) concentration in the electrolyte and includes a single ODE in each electrode for the average particle concentration. Options can be passed to allow different particle models, but there is only ever a single particle in each electrode (i.e. we solve for the x-averaged concentration).

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/master/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/models/submodels/electrolyte_diffusion/composite_diffusion.py [hunk 1]
```diff
@@ -95,14 +95,3 @@ def set_initial_conditions(self, variables):
         c_e = variables["Electrolyte concentration"]
 
         self.initial_conditions = {c_e: self.param.c_e_init}
-
-    def set_boundary_conditions(self, variables):
```
Comment (tinosulzer) [inline pybamm/models/submodels/electrolyte_diffusion/composite_diffusion.py:99]: I think at some point we moved these into the submodels to be clearer what the submodels are. It's not too much repeated code.

File: tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_dfn.py [hunk 2]
```diff
@@ -168,6 +168,10 @@ def positive_radius(x):
 
         param["Negative particle radius [m]"] = negative_radius
         param["Positive particle radius [m]"] = positive_radius
+        # Only get 3dp of accuracy in some tests at 1C with particle distribution
+        # TODO: investigate if there is a bug or some way to improve the
+        # implementation
```
Comment (tinosulzer) [inline tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_dfn.py:173]: is this related to what @brosaplanella was seeing?

**Review by tinosulzer — APPROVED** [review]
Looks great, thanks!

File: pybamm/models/submodels/electrolyte_diffusion/composite_diffusion.py (hunk 1 above)
Comment (rtimms) [reply pybamm/models/submodels/electrolyte_diffusion/composite_diffusion.py:99]: Makes sense, I will put them back. I did it this way as you also need to provide boundary conditions for the constant concentration model so that `grad(c_e)` ends up having the correct shape.

File: tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_dfn.py (hunk 2 above)
Comment (rtimms) [reply tests/integration/test_models/test_full_battery_models/test_lithium_ion/test_dfn.py:177]: looks like it. you can get 4dp of accuracy everywhere else.
